nmosregistration/aggregation.py

```diff
@@ -38,23 +39,33 @@
 
 class AggregatorAPI(WebAPI):
 
-    def __init__(self, logger, config, registry=EtcdInterface()):
+    def __init__(self, logger, config):
+        if config['registry']['type'] == 'couchbase':
+            registry = CouchbaseInterface(
+                cluster_address=config['registry']['hosts'],
+                username=config['registry']['username'],
+                password=config['registry']['password'],
+                bucket=config['registry']['bucket']
+            )
+        else:
+            registry = EtcdInterface()
         super(AggregatorAPI, self).__init__()
         self._config = config
 
-        garbage_collect_interval = int(self._config.get("garbage_collect_interval", 10))
-        self._garbage_collector = GarbageCollect(identifier=HOST, registry=registry, interval=garbage_collect_interval)
+        if config['registry']['type'] == 'etcd':
+            garbage_collect_interval = int(self._config.get("garbage_collect_interval", 10))
```

Comment:
Could this be definition be moved into the config file and the start of the service file?